### PR TITLE
Fix#11814: Support DefaultCommand for submit button outside (but linked to) form

### LINKED ISSUE #11814: DefaultCommand: Not working for button outside (but linked to) form
### Describe the bug

The DefaultCommand component only works when it targets CommandButtons that are linked to (using the form-attribute), but placed outside the form. See reproducer.

I'm not entirely sure if this should be classified as a bug or not, but I thought it would make sense if this usage would be supported (it sure would help in my use-case).

I already looked through the code and noticed that it indeed doesn't take into account buttons that are not placed inside the form, it only takes into account the form that is actualy enclosed by the target button: https://github.com/primefaces/primefaces/blob/3c1945e926507fe6831c414b962e31d68de47002/primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.defaultcommand.js#L40

If possible I'd like to contribute with a fix for this!

### Reproducer

In the reproducer I created a form with an InputText, CommandButton "Submit inside form" and DefaultCommand. Next to the form there's another CommandButton "Submit outside form" that is linked to the form using the form-attribute. I configured the DefaultCommand to point to button "Submit outside form", but when I press the Enter-key when my focus is in the InputText, the form is submitted using button "Submit inside form".

[PrimeFaces_DefaultCommand_reproducer.zip](https://github.com/primefaces/primefaces/files/15113064/PrimeFaces_DefaultCommand_reproducer.zip)

Relevant portion from reproducer:
```xml
<h:form id="frmTest">
    <p:inputText value="#{testView.inputValue}" />
    <p:commandButton id="buttonInsideForm" value="Submit inside form" action="#{testView.submitActionFromInsideForm}" />
    <p:defaultCommand target="buttonOutsideForm" />
</h:form>

<p:commandButton id="buttonOutsideForm" value="Submit outside form" action="#{testView.submitActionFromOutsideForm}" form="frmTest" />
```

### Expected behavior

When pressing the enter key in the InputText, the form should be submitted by button "Submit outside form", as configured in the DefaultCommand component.

### PrimeFaces edition

Community

### PrimeFaces version

13.0.8

### Theme

_No response_

### JSF implementation

All

### JSF version

2.3

### Java version

17

### Browser(s)

_No response_

### PR DESCRIPTION
**Pull request by BjornJaspers**
Fix #11814 

This is how I thought about fixing this, but of course I'm open for discussion.

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
This actually looks really reasonable to me.

**Comment by tandraschko** [comment]
what about normal button, linkbutton or commandlink?

**Comment by BjornJaspers** [comment]
> what about normal button, linkbutton or commandlink?

I updated the implementation to support commandlink. Referencing a form in a normal button and linkbutton doesn't seem to be supported however.

**Comment by melloware** [comment]
Agreed this would only work from CommandXXX types thanks for adding CommandLink

File: primefaces/src/main/java/org/primefaces/component/commandlink/CommandLinkRenderer.java [hunk 1]
```diff
@@ -72,6 +72,7 @@ protected void encodeMarkup(FacesContext context, CommandLink link) throws IOExc
         ResponseWriter writer = context.getResponseWriter();
         String clientId = link.getClientId(context);
         Object label = link.getValue();
+        String referencedForm = link.getForm();
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/commandlink/CommandLinkRenderer.java]: i would just call the var "form"

**Comment by tandraschko** [comment]
+1 then

**Comment by melloware** [comment]
@BjornJaspers unfortunately this fix cannot be merged in the 13.X branch that you have it marked for.  Can you resubmit this under branch `master` and leave it up to us to backport to 14.X.   13.X is now under LTS by PrimeTek so only PRO users will received fixes on 13.X

**Pull request closed without merging**